processing/src/test/java/io/druid/segment/filter/LikeFilterTest.java

```diff
@@ -0,0 +1,197 @@
+      PARSER.parse(ImmutableMap.<String, Object>of("dim0", "1", "dim1", "foo")),
+      PARSER.parse(ImmutableMap.<String, Object>of("dim0", "2", "dim1", "foobar")),
+      PARSER.parse(ImmutableMap.<String, Object>of("dim0", "3", "dim1", "bar")),
+      PARSER.parse(ImmutableMap.<String, Object>of("dim0", "4", "dim1", "foobarbaz")),
+      PARSER.parse(ImmutableMap.<String, Object>of("dim0", "5", "dim1", "foo%bar"))
+  );
+
+  public LikeFilterTest(
+      String testName,
+      IndexBuilder indexBuilder,
+      Function<IndexBuilder, Pair<StorageAdapter, Closeable>> finisher,
+      boolean optimize
+  )
+  {
+    super(testName, ROWS, indexBuilder, finisher, optimize);
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception
+  {
+    BaseFilterTest.tearDown(LikeFilterTest.class.getName());
+  }
+
+  @Test
+  public void testExactMatch()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "bar", null, null),
+        ImmutableList.of("3")
+    );
+  }
+
+  @Test
+  public void testExactMatchWithEscape()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "@bar", "@", null),
+        ImmutableList.of("3")
+    );
+  }
+
+  @Test
+  public void testExactMatchWithExtractionFn()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "bar", null, new SubstringDimExtractionFn(3, 3)),
+        ImmutableList.of("2", "4")
+    );
+  }
+
+  @Test
+  public void testPrefixMatch()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "foo%", null, null),
+        ImmutableList.of("1", "2", "4", "5")
+    );
+  }
+
+  @Test
+  public void testPrefixMatchWithEscape()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "foo@%%", "@", null),
+        ImmutableList.of("5")
+    );
+  }
+
+  @Test
+  public void testPrefixMatchWithExtractionFn()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "a%", null, new SubstringDimExtractionFn(1, null)),
+        ImmutableList.of("3")
+    );
+  }
+
+  @Test
+  public void testWildcardMatch()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "%oba%", null, null),
+        ImmutableList.of("2", "4")
+    );
+  }
+
+  @Test
+  public void testWildcardMatchWithEscape()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "%@%ba%", "@", null),
+        ImmutableList.of("5")
+    );
+  }
+
+  @Test
+  public void testPrefixAndSuffixMatch()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "f%r", null, null),
+        ImmutableList.of("2", "5")
+    );
+  }
+
+  @Test
+  public void testUnderscoreMatch()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "f_o", null, null),
+        ImmutableList.of("1")
+    );
+  }
+
+  @Test
+  public void testSuffixMatchWithExtractionFn()
+  {
+    assertFilterMatches(
+        new LikeDimFilter("dim1", "%ar", null, new SubstringDimExtractionFn(3, 3)),
+        ImmutableList.of("2", "4")
+    );
+  }
+
```

Comment:
minor nit: can we also add boundary tests for matchAll, matchNone, match against string having wildcard only ?